Tasks/AzureRmWebAppDeployment/azurermutil.ts

```diff
@@ -129,25 +131,30 @@ function getAuthorizationToken(SPN): Q.Promise<string> {
     return deferred.promise;
 }
 
-async function getAzureRMWebAppID(SPN, webAppName: string, resourceType: string) {
+async function getAzureRMWebAppID(SPN, webAppName: string,url: string) {
 
     var deferred = Q.defer<any>();
     var accessToken = await getAuthorizationToken(SPN);
-
-    var url = armUrl + 'subscriptions/' + SPN.subscriptionId + '/resources?$filter=resourceType EQ \'' + resourceType +
-                        '\' AND name EQ \'' + webAppName + '\'&api-version=2016-07-01';
     var headers = {
         authorization: 'Bearer '+ accessToken
     };
 
     tl.debug('Requesting AzureRM Web App ID: ' + url);
-    httpObj.get('GET', url, headers, (error, response, body) => {
+    httpObj.get('GET', url, headers, async (error, response, body) => {
         if(error) {
             deferred.reject(error);
         }
         else if(response.statusCode === 200) {
             var webAppIDDetails = JSON.parse(body);
-            deferred.resolve(webAppIDDetails.value[0]);
+            if(webAppIDDetails.nextLink){
```

Comment:
Should you attempt to query next link only when (webAppIDDetails.value.length == 0) && (webAppIDDetails.nextLink)?. If you are able to get value in the first response itself, why make query to next page?